metals/src/main/scala/scala/meta/internal/worksheets/WorksheetProvider.scala

```diff
@@ -434,9 +434,13 @@ class WorksheetProvider(
         }
         if isSupported
       } yield {
+        // We filter out NonUnitStatements from wartremover or you'll get an
+        // error about $doc.binder returning Unit from mdoc, which causes the
+        // worksheet not to be evaluated.
         val scalacOptions = info.scalac.getOptions.asScala
           .filterNot(_.contains("semanticdb"))
           .filterNot(_.contains("-Wconf"))
+          .filterNot(_.contains("org.wartremover.warts.NonUnitStatements"))
```

Comment:
maybe drop all that contains `wartremover` ?

Reply:
I was thinking that, but I thought this would maybe cause the least amount of confusions. I'm still not 100% sure tbh.

Reply:
I don't know. I took a quick look to check how options is passed for wartremover.
I'm not sure, but in documentation it looks like a single option where rules are separated with `:`. Am I right?
In this case it should be equal to removing all wartremover configuration if `NonUnitStatements` is enabled.

Reply:
No, it ends up being split up in the `scalacOptions` like this:

```
-Xplugin:target/compiler_plugins/wartremover_2.13.8-3.0.4.jar, -P:wartremover:traverser:org.wartremover.warts.Any, -P:wartremover:traverser:org.wartremover.warts.AsInstanceOf, -P:wartremover:traverser:org.wartremover.warts.DefaultArguments, -P:wartremover:traverser:org.wartremover.warts.EitherProjectionPartial, -P:wartremover:traverser:org.wartremover.warts.IsInstanceOf, -P:wartremover:traverser:org.wartremover.warts.IterableOps, -P:wartremover:traverser:org.wartremover.warts.NonUnitStatements, -P:wartremover:traverser:org.wartremover.warts.Null, -P:wartremover:traverser:org.wartremover.warts.OptionPartial, -P:wartremover:traverser:org.wartremover.warts.Product, -P:wartremover:traverser:org.wartremover.warts.Return, -P:wartremover:traverser:org.wartremover.warts.Serializable, -P:wartremover:traverser:org.wartremover.warts.StringPlusAny, -P:wartremover:traverser:org.wartremover.warts.Throw, -P:wartremover:traverser:org.wartremover.warts.TryPartial, -P:wartremover:traverser:org.wartremover.warts.Var```
So every rule is passed in as a flag.

Reply:
ok, thanks for explanation!
sorry got the wrong assumption